### DISCUSSION TITLE
Support for more recent versions of TCA?

### DISCUSSION THREAD

**Discussion by tw0po1nt** [pointfreeco/composable-core-location#34] [General]
I am currently building an app in TCA to keep my chops fresh on the latest in Apple Platforms development. The app needs to hook into CoreLocation, and hence I found this repository. But just from searching, it appears that this library has not been updated to support newer versions of TCA (updated concurrency, reducer protocol, etc).

Are there plans to update this library? Or can anyone direct me to an example of implementing the delegation pattern of CoreLocation (and many other Apple APIs) within newer versions of TCA? The closest example I could find was in the audio recorder case study.

**Comment by mbrandonw** [comment]
Hi @tw0po1nt, thanks for bringing this up.

We do plan on updating this library eventually, but thanks to `async` and `AsyncStream` in Swift, we can even fully disentangle it from TCA. No need to make use of `Effect`s. That would make the library useful even for vanilla SwiftUI/UIKit applications.

That is the direction we want to go, but we also want it to play nicely with the dependency management system in TCA. So, we plan on breaking that piece into its own library (very soon), and then we will be in a better position to update this library.

**Comment by mbrandonw** [comment]
Also, I'm going to convert this to a discussion.

**Comment by teleki** [comment]
After today’s episode I got all excited and updated my project to the `prerelease/1.0` version of TCA. Unfortunately, I immediately got 61 issues with Composable Core Location. 

Do you have any tips on how to proceed in a situation like this? Is it best to wait on both to be updated? I was going to attempt the update, but then I realized that you are also working on async/await updates, and that is not something that I can undertake.

Your tips and suggestions are welcome!

**Reply by 3a4oT** [reply]
Hey, I recently opened an issue about it. https://github.com/pointfreeco/composable-core-location/issues/36 cc @mbrandonw
